reactor-core/src/main/java/reactor/core/publisher/FluxSwitchOnFirst.java

```diff
@@ -19,610 +19,1029 @@
+						this.transformer.apply(Signal.next(t, o.currentContext()), this),
+						"The transformer returned a null value"
+					);
+				}
+				catch (Throwable e) {
+					this.done = true;
+					o.onError(Operators.onOperatorError(this.s, e, t, o.currentContext()));
+					return;
+				}
+
+				result.subscribe(o);
+				return;
+			}
+
+			this.inner.onNext(t);
+		}
+
+		@Override
+		public void onError(Throwable t) {
+			if (this.done) {
+				Operators.onErrorDropped(t, this.outer.currentContext());
+				return;
+			}
+
+			this.done = true;
+			this.throwable = t;
+
+			final long previousState = markInboundTerminated(this);
+			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
+				Operators.onErrorDropped(t, this.outer.currentContext());
+				return;
+			}
+
+			if (hasFirstValueSent(previousState)) {
+				this.inner.onError(t);
+				return;
+			}
+
+			if (!hasFirstValueReceived(previousState)) {
+				final Publisher<? extends R> result;
+				final CoreSubscriber<? super R> o = this.outer;
+				try {
+					result = Objects.requireNonNull(this.transformer.apply(Signal.error(t,
+							o.currentContext()), this),
+							"The transformer returned a null value");
+				}
+				catch (Throwable e) {
+					o.onError(Exceptions.addSuppressed(e, t));
+					return;
+				}
+
+				result.subscribe(o);
+			}
+		}
+
+		@Override
+		public void onComplete() {
+			if (this.done) {
+				return;
+			}
+
+			this.done = true;
+
+			final long previousState = markInboundTerminated(this);
+			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
+				return;
+			}
+
+			if (hasFirstValueSent(previousState)) {
+				this.inner.onComplete();
+				return;
+			}
+
+			if (!hasFirstValueReceived(previousState)) {
+				final Publisher<? extends R> result;
+				final CoreSubscriber<? super R> o = this.outer;
+
+				try {
+					result =
+							Objects.requireNonNull(this.transformer.apply(Signal.complete(
+									o.currentContext()), this),
+									"The transformer returned a null value");
+				}
+				catch (Throwable e) {
+					o.onError(e);
+					return;
+				}
+
+				result.subscribe(o);
+			}
+		}
+
+		@Override
+		public void cancel() {
+			long previousState = markInboundCancelled(this);
+			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
+				return;
+			}
+
+			this.s.cancel();
+
+			if (hasFirstValueReceived(previousState) && !hasInboundRequestedOnce(previousState)) {
+				final T f = this.firstValue;
+				Operators.onDiscard(f, currentContext());
+			}
+		}
+
+		void cancelAndError() {
+			long previousState = markInboundCancelled(this);
+			if (hasInboundCancelled(previousState) || hasInboundTerminated(previousState)) {
+				return;
+			}
+
+			this.s.cancel();
+
+			if (hasFirstValueReceived(previousState) && !hasInboundRequestedOnce(previousState)) {
+				final T f = this.firstValue;
+				Operators.onDiscard(f, currentContext());
```

Comment:
same as above, consider nulling out `this.firstValue` to avoid object retention?